1.10.x Smoke test failure TestBindingWithImage

```
--- FAIL: TestBindingWithImage (2.54s)
    --- FAIL: TestBindingWithImage/run_with_initial_image (2.06s)
panic: Asynchronous assertion has a GomegaMatcher as the first element of optionalDescription.
	Do you mean to use And/Or/SatisfyAll/SatisfyAny to combine multiple matchers? [recovered]
	panic: Asynchronous assertion has a GomegaMatcher as the first element of optionalDescription.
	Do you mean to use And/Or/SatisfyAll/SatisfyAny to combine multiple matchers?

goroutine 15793 [running]:
testing.tRunner.func1.2({0x1ece000, 0xc000305f00})
	/opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1209 +0x24e
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1212 +0x218
panic({0x1ece000, 0xc000305f00})
	/opt/hostedtoolcache/go/1.17.13/x64/src/runtime/panic.go:1038 +0x215
github.com/onsi/gomega/internal.vetOptionalDescription({0x22f165a, 0x16}, {0xc000305db0, 0x40f3a7, 0x10})
	/home/runner/go/pkg/mod/github.com/onsi/gomega@v1.20.1/internal/vetoptdesc.go:18 +0xb9
github.com/onsi/gomega/internal.(*AsyncAssertion).Should(0xc0003f9180, {0x264b548, 0xc000981cc0}, {0xc000305db0, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/onsi/gomega@v1.20.1/internal/async_assertion.go:107 +0x65
github.com/apache/camel-k/e2e/global/common.TestBindingWithImage.func1.1(0x752072656e696174)
	/home/runner/work/camel-k/camel-k/e2e/global/common/kamelet_binding_with_image_test.go:82 +0x555
testing.tRunner(0xc00040f040, 0xc0007dd560)
	/opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.17.13/x64/src/testing/testing.go:1306 +0x35a
FAIL	github.com/apache/camel-k/e2e/global/common	688.253s
FAIL
```
It happens only in `release-1.10.x`, likely it's some upgraded dependency.

--- a/e2e/global/common/kamelet_binding_with_image_test.go
+++ b/e2e/global/common/kamelet_binding_with_image_test.go
@@ -23,14 +23,14 @@ limitations under the License.
 package common
 
 import (
-	"github.com/onsi/gomega/gstruct"
 	"testing"
 
-	. "github.com/apache/camel-k/e2e/support"
 	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gstruct"
 
 	corev1 "k8s.io/api/core/v1"
 
+	. "github.com/apache/camel-k/e2e/support"
 	"github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
 )
 
@@ -80,8 +80,9 @@ func TestBindingWithImage(t *testing.T) {
 			Eventually(IntegrationGeneration(ns, bindingID)).
 				Should(gstruct.PointTo(BeNumerically("==", 1)))
 			Eventually(Integration(ns, bindingID)).Should(WithTransform(Annotations, And(
-				HaveKeyWithValue("test", "1"))),
-				HaveKeyWithValue("trait.camel.apache.org/container.image", expectedImage))
+				HaveKeyWithValue("test", "1"),
+				HaveKeyWithValue("trait.camel.apache.org/container.image", expectedImage),
+			)))
 			Eventually(IntegrationStatusImage(ns, bindingID)).
 				Should(Equal(expectedImage))
 			Eventually(IntegrationPodPhase(ns, bindingID), TestTimeoutLong).
@@ -100,8 +101,9 @@ func TestBindingWithImage(t *testing.T) {
 			Eventually(IntegrationGeneration(ns, bindingID)).
 				Should(gstruct.PointTo(BeNumerically("==", 1)))
 			Eventually(Integration(ns, bindingID)).Should(WithTransform(Annotations, And(
-				HaveKeyWithValue("test", "2"))),
-				HaveKeyWithValue("trait.camel.apache.org/container.image", expectedImage))
+				HaveKeyWithValue("test", "2"),
+				HaveKeyWithValue("trait.camel.apache.org/container.image", expectedImage),
+			)))
 			Eventually(IntegrationStatusImage(ns, bindingID)).
 				Should(Equal(expectedImage))
 			Eventually(IntegrationPodPhase(ns, bindingID), TestTimeoutLong).

